src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -4317,13 +4317,26 @@ public override bool PreVisit(CallNode node)
                         _txb.AddVolatileVariables(args[i], volatileVariables);
                     }
 
+                    var isIdentifier =
+                        args[i] is FirstNameNode &&
+                        _features.HasFlag(Features.SupportIdentifiers) &&
+                        maybeFunc.IsIdentifierParam(i);
+
                     // Use the new scope only for lambda args.
                     _currentScope = (maybeFunc.IsLambdaParam(i) && scopeInfo.AppliesToArgument(i)) ? scopeNew : scopeNew.Parent;
-                    args[i].Accept(this);
 
-                    _txb.AddVolatileVariables(node, _txb.GetVolatileVariables(args[i]));
+                    if (!isIdentifier)
+                    {
+                        args[i].Accept(this);
+                        _txb.AddVolatileVariables(node, _txb.GetVolatileVariables(args[i]));
+                        argTypes[i] = _txb.GetType(args[i]);
+                    }
+                    else
+                    {
+                        // This is an identifier, no associated type 
+                        argTypes[i] = DType.Unknown;
```

Comment:
>DType.Unknown [](http://example.com/codeflow?start=38&length=13)

I think this is fine - since we should never actually use the type.  (ie, curious - could we set it to null and not hit anything?) 

We could also "poison pill" it by making it error.

Reply:
I think Unknown is the most appropriate here
- null will immediately generate a NRE at Contracts.Assert(argTypes[i].IsValid)
- DType.Error will make it fail on "check.ThrowOnErrors();" a bit later in the execution

Notice that the real type of the new column will be determined later when calling maybeFunc.HandleCheckInvocation(_.., out var returnType, ...);

Reply:
"I think Unknown is the most appropriate here" 
We need a stronger argument, especially because, despite the name, Unknown's semantics are somewhat ambiguous (we hit this exact issue on the deferred work for #720 ). 

If other code is actually reading this value - then it's possible that assigning it Unknown could create bugs. 

Since nobody should actually be reading this value, we should look at the exact locations that are reading this -  and understand why. Should they not be?

Reply:
You're right.
I've set it to null and verified where it's used.
There are only 2 places:
- right after this code in Contracts.Assert
- in TexfFuntion.CheckTypesCore where we can easily avoid the problem